library/sinks/FileSystem.ts

```diff
@@ -4,50 +4,46 @@ import { InterceptorResult } from "../agent/hooks/MethodInterceptor";
 
-const functionsWithPath = [
-  "access",
-  "appendFile",
-  "chmod",
-  "chown",
-  "createReadStream",
-  "createWriteStream",
-  "exists",
-  "lchmod",
-  "lchown",
-  "lutimes",
-  "lstat",
-  "mkdir",
-  "open",
-  "openAsBlob",
-  "opendir",
-  "readdir",
-  "readFile",
-  "readlink",
-  "unlink",
-  "realpath",
-  "rename",
-  "rmdir",
-  "rm",
-  "symlink",
-  "stat",
-  "statfs",
-  "truncate",
-  "utimes",
-  "writeFile",
-  "copyFile",
-  "cp",
-];
-
-const noSync = ["createReadStream", "createWriteStream", "openAsBlob"];
-
-const noPromise = [
-  "createReadStream",
-  "createWriteStream",
-  "exists",
-  "openAsBlob",
-];
-
-const withSecondPathArgument = ["copyFile", "cp", "link", "rename", "symlink"];
+type FileSystemFunction = {
+  pathsArgs: number; // The amount of arguments that are paths
+  sync: boolean; // Whether the function has a synchronous version (e.g. fs.accessSync)
+  promise: boolean; // Whether the function has a promise version (e.g. fs.promises.access)
+};
+
+const functions: Record<string, FileSystemFunction> = {
+  access: { pathsArgs: 1, sync: true, promise: true },
+  appendFile: { pathsArgs: 1, sync: true, promise: true },
+  chmod: { pathsArgs: 1, sync: true, promise: true },
+  chown: { pathsArgs: 1, sync: true, promise: true },
+  createReadStream: { pathsArgs: 1, sync: false, promise: false },
+  createWriteStream: { pathsArgs: 1, sync: false, promise: false },
+  exists: { pathsArgs: 1, sync: true, promise: false },
+  lchmod: { pathsArgs: 1, sync: true, promise: true },
+  lchown: { pathsArgs: 1, sync: true, promise: true },
+  lutimes: { pathsArgs: 1, sync: true, promise: true },
+  lstat: { pathsArgs: 1, sync: true, promise: true },
+  mkdir: { pathsArgs: 1, sync: true, promise: true },
+  open: { pathsArgs: 1, sync: true, promise: true },
+  openAsBlob: { pathsArgs: 1, sync: false, promise: false },
+  opendir: { pathsArgs: 1, sync: true, promise: true },
+  readdir: { pathsArgs: 1, sync: true, promise: true },
+  readFile: { pathsArgs: 1, sync: true, promise: true },
+  readlink: { pathsArgs: 1, sync: true, promise: true },
+  unlink: { pathsArgs: 1, sync: true, promise: true },
+  realpath: { pathsArgs: 1, sync: true, promise: true },
+  rename: { pathsArgs: 2, sync: true, promise: true },
+  rmdir: { pathsArgs: 1, sync: true, promise: true },
+  rm: { pathsArgs: 1, sync: true, promise: true },
+  symlink: { pathsArgs: 2, sync: true, promise: true },
+  stat: { pathsArgs: 1, sync: true, promise: true },
+  statfs: { pathsArgs: 1, sync: true, promise: true },
+  truncate: { pathsArgs: 1, sync: true, promise: true },
+  utimes: { pathsArgs: 1, sync: true, promise: true },
+  writeFile: { pathsArgs: 1, sync: true, promise: true },
+  copyFile: { pathsArgs: 2, sync: true, promise: true },
+  cp: { pathsArgs: 2, sync: true, promise: true },
+  link: { pathsArgs: 2, sync: true, promise: true },
+};
```

Comment:
Maybe add [glob](https://nodejs.org/api/fs.html#fspromisesglobpattern-options), [watch](https://nodejs.org/api/fs.html#fspromiseswatchfilename-options) and [mkdtemp](https://nodejs.org/api/fs.html#fspromisesmkdtempprefix-options)?